### PR TITLE
Fixes #29212: In JS, completion won't suggest TypeScript Syntax

### LINKED ISSUE #29212: In JS, completion list suggests TypeScript syntax
**TypeScript Version:**  3.3.0-dev.20181222

**Search Terms:**

**Code**

```ts
// trigger completion list anywhere in a JS file
```

**Expected behavior:**

Completion list doesn't contain TypeScript specific syntax in JS files.

**Actual behavior:**

In JS files it suggests `enum`, `interface`, `namespace`, `type`, `declare`, `abstract`, `readonly`, `public`, `protected`, `private`.

**Playground Link:** 

**Related Issues:**

### PR DESCRIPTION
**Pull request by suhailsinghbains**
Fixes #29212 
* [Y] There is an associated issue that is labeled
  'Bug' or 'help wanted' or is in the Community milestone
* [Y] Code is up-to-date with the `master` branch
* [N] You've successfully run `jake runtests` locally
* [N] You've signed the CLA
* [N] There are new or updated unit tests validating the change

I had gone through [0] link, and implement the changes accordingly.
I wasn't able to add test, I might some extra help :(

0: https://github.com/Microsoft/TypeScript/pull/21257/files

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -953,6 +953,7 @@ namespace ts.Completions {
                 || tryGetConstructorCompletion()
                 || tryGetClassLikeCompletionSymbols()
                 || tryGetJsxCompletionSymbols()
+                || JSFileType()
                 || (getGlobalCompletions(), GlobalsSearch.Success);
```
Comment (sheetalkamat) [inline src/services/completions.ts]: Don't think you want to add here. Instead you want to set the filter in `getGlobalCompletions` by checking  `isSourceFileJS(sourceFile)`

File: src/services/completions.ts [hunk 2]
```diff
@@ -1933,6 +1939,8 @@ namespace ts.Completions {
                     return isFunctionLikeBodyKeyword(kind);
                 case KeywordCompletionFilters.TypeKeywords:
                     return isTypeKeyword(kind);
+                case KeywordCompletionFilters.JSFileFilter:
```
Comment (sheetalkamat) [inline src/services/completions.ts]: You want to actually filter out non typescript keywords here

**Comment by suhailsinghbains** [comment]
Hi @sheetalkamat 
Thanks for the review, I have added the changes requested.
I would need some help for adding a test, please
Thanks

**Comment by RyanCavanaugh** [comment]
@suhailsinghbains thanks for the effort in making a contribution - unfortunately we're not staffed to provide extensive support on external PRs. There are a lot of historical PRs you can look through for guidance if that'd be useful. Without appropriate testcases as well as  passing CI, we won't be able to provide additional support here. Thanks for understanding!

**Pull request closed without merging**